.github/workflows/build-toolchain.yml

```diff
@@ -90,43 +96,73 @@ jobs:
           "INCLUDE=$MSVCDir\ATL\INCLUDE;$MSVCDir\INCLUDE;$MSVCDir\MFC\INCLUDE;$env:INCLUDE" >> $env:GITHUB_ENV
           "LIB=$MSVCDir\LIB;$MSVCDir\MFC\LIB;$env:LIB" >> $env:GITHUB_ENV
 
-      - name: Build ${{ inputs.game }} with CMake using ${{ inputs.preset }} preset
+      - name: Set Up VC2022 Environment
+        if: startsWith(inputs.preset, 'win32')
+        uses: ilammy/msvc-dev-cmd@v1
+        with:
+          arch: x86
+
+      - name: Configure ${{ inputs.game }} with CMake Using ${{ inputs.preset }}${{ inputs.tools && '+t' || '' }}${{ inputs.extras && '+e' || '' }} Preset
         shell: pwsh
         run: |
-          Write-Host "Configuring project with CMake using preset: ${{ inputs.game }} (${{ inputs.preset }})"
-
-          # Set build flags based on game and tools
           if ("${{ inputs.game }}" -eq "Generals") {
-            $buildFlags = @("-DGENZH_BUILD_ZEROHOUR=OFF", "-DGENZH_BUILD_GENERALS=ON")
-            if ("${{ inputs.tools }}" -eq "true") {
-              $buildFlags += "-DGENZH_BUILD_GENERALS_TOOLS=ON"
-              $buildFlags += "-DGENZH_BUILD_GENERALS_EXTRAS=ON"
+            $buildFlags = @(
+              "-DGENZH_BUILD_ZEROHOUR=OFF",
+              "-DGENZH_BUILD_GENERALS=ON",
+              "-DGENZH_BUILD_GENERALS_TOOLS=ON",
+              "-DGENZH_BUILD_GENERALS_EXTRAS=OFF"
+            )
+
+            if ("${{ inputs.tools }}" -eq "false") {
+              $buildFlags[2] = "-DGENZH_BUILD_GENERALS_TOOLS=OFF"
             }
-          } else {
-            $buildFlags = @("-DGENZH_BUILD_ZEROHOUR=ON", "-DGENZH_BUILD_GENERALS=OFF")
-            if ("${{ inputs.tools }}" -eq "true") {
-              $buildFlags += "-DGENZH_BUILD_ZEROHOUR_TOOLS=ON"
-              $buildFlags += "-DGENZH_BUILD_ZEROHOUR_EXTRAS=ON"
+
+            if ("${{ inputs.extras }}" -eq "true") {
+              $buildFlags[3] = "-DGENZH_BUILD_GENERALS_EXTRAS=ON"
+            }
+          } elseif ("${{ inputs.game }}" -eq "GeneralsMD") {
+            $buildFlags = @(
+              "-DGENZH_BUILD_ZEROHOUR=ON",
+              "-DGENZH_BUILD_GENERALS=OFF",
+              "-DGENZH_BUILD_ZEROHOUR_TOOLS=ON",
+              "-DGENZH_BUILD_ZEROHOUR_EXTRAS=OFF"
+            )
+
+            if ("${{ inputs.tools }}" -eq "false") {
+              $buildFlags[2] = "-DGENZH_BUILD_ZEROHOUR_TOOLS=OFF"
```

Comment:
This indexing looks error prone.

How about
```
if ("${{ inputs.tools }}" -eq "true") {
  $buildFlags += "-DGENZH_BUILD_ZEROHOUR_TOOLS=ON"
} else {
  $buildFlags += "-DGENZH_BUILD_ZEROHOUR_TOOLS=OFF"
}
```

Reply:
yeah

Reply:
the same for extras?